toolchain.mk

```diff
@@ -56,17 +56,38 @@ AARCH64_CROSS_COMPILE 		?= $(AARCH64_PATH)/bin/aarch64-linux-gnu-
 AARCH64_GCC_VERSION 		?= arm-gnu-toolchain-11.3.rel1-x86_64-aarch64-none-linux-gnu
 SRC_AARCH64_GCC 		?= https://developer.arm.com/-/media/Files/downloads/gnu/11.3.rel1/binrel/$(AARCH64_GCC_VERSION).tar.xz
 
+RUST_TOOLCHAIN_PATH 		?= $(TOOLCHAIN_ROOT)/rust
+
 .PHONY: toolchains
-toolchains: aarch32 aarch64
+toolchains: aarch32-toolchain aarch64-toolchain rust-toolchain
 
-.PHONY: aarch32
-aarch32:
+.PHONY: aarch32-toolchain
+aarch32-toolchain:
 	$(call dltc,$(AARCH32_PATH),$(SRC_AARCH32_GCC),$(AARCH32_GCC_VERSION))
 
-.PHONY: aarch64
-aarch64:
+.PHONY: aarch64-toolchain
+aarch64-toolchain:
 	$(call dltc,$(AARCH64_PATH),$(SRC_AARCH64_GCC),$(AARCH64_GCC_VERSION))
 
+# Download the Rust toolchain
+define dl-rust-toolchain
+	@if [ ! -d "$(1)" ]; then \
+		( mkdir -p $(1) && \
+		export RUSTUP_HOME=$(1)/.rustup && \
+		export CARGO_HOME=$(1)/.cargo && \
+		curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path && \
+		source "$(1)/.cargo/env" && \
+		rustup target install aarch64-unknown-linux-gnu && \
+		rustup target install arm-unknown-linux-gnueabihf && \
+		rustup default nightly-2023-12-18 ) || \
+		( echo Rust toolchain installation failed && rm -rf $(1) ); \
```

Comment:
The infamous `rm -rf`, what if `RUST_TOOLCHAIN_PATH` has been set to something bad, like `/` or `$(HOME)`?

Reply:
...then the user is left homeless! :rofl: 
I agree. In fact the whole `||` part is not needed IMO. If something fails here, better let the user clean up :wink:

Reply:
Yeah but I was following the existing practice: https://github.com/OP-TEE/build/blob/master/toolchain.mk#L22. However, I can `||` part.

Reply:
> Yeah but I was following the existing practice: https://github.com/OP-TEE/build/blob/master/toolchain.mk#L22. However, I can drop `||` part.

Ouch, people for some reason seem to love `rm -rf`.